opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1000,21 +1001,32 @@ namespace Opm {
                 fip_.fip[i].resize(nc,0.0);
             }
 
-            for (int c = 0; c < nc; ++c) {
-                const auto& intQuants = *ebosSimulator_.model().cachedIntensiveQuantities(c, /*timeIdx=*/0);
+            ElementContext elemCtx(ebosSimulator_);
+            auto elemIt = elemCtx.gridView().template begin</*codim=*/0>();
+            const auto& elemEndIt = elemCtx.gridView().template end</*codim=*/0>();
+            for (; elemIt != elemEndIt; ++elemIt) {
+                const auto& elem = *elemIt;
+                if (elem.partitionType() != Dune::InteriorEntity)
+                    continue;
+
+                elemCtx.updatePrimaryStencil(elem);
```

Comment:
I have to admit this part is not very intuitive... (I'm just looping over the cells, why do I need to update a stencil?!?) I'll accept the code though now that I understand what it does (but wishing for improvements in the future).